src/panda_eup/pbd_interface.py

```diff
@@ -35,95 +36,117 @@ def __init__(self):
-                                                         EnableTeaching)
-        except rospy.ServiceException:
-            rospy.logerr('Cannot create Kinesthetic Teaching client!')
-
-        try:
-            self.kinesthetic_client.wait_for_service(5.0)
-        except rospy.ROSException:
-            rospy.logerr('Cannot contact the Primitive Interface Node!')
-
-        self.interpreter = interpreter.PandaProgramInterpreter()  # internal interpreter, for execute_now_primitives
-
+        if not self.robotless_debug:
+            self.gripper_state_subscriber = rospy.Subscriber("/franka_gripper/joint_states", JointState,
+                                                            self.gripper_state_callback)
+
+            try:
+                self.kinesthetic_client = rospy.ServiceProxy('/primitive_interface_node/kinesthetic_teaching',
+                                                            EnableTeaching)
+            except rospy.ServiceException:
+                rospy.logerr('Cannot create Kinesthetic Teaching client!')
+
+            try:
+                self.kinesthetic_client.wait_for_service(5.0)
+            except rospy.ROSException:
+                rospy.logerr('Cannot contact the Primitive Interface Node!')
+
+        #The interpreter of panda_widgets is used for the tuning and running of existing programs, the interpreter of pbd_interface
+        #is used for actions that are targeted at the program-to-be-created such as moving to previous preconditions when primitives are deleted
+        self.interpreter = interpreter.PandaProgramInterpreter(robotless_debug=self.robotless_debug)  # internal interpreter, for execute_now_primitives
+        self.interpreter.loaded_program = self.program
         self.freeze()
 
     def initialize_program(self):
         # TODO: maybe enforce some check on the rest of the functions? to check for initialization
-        while self.last_pose is None or self.last_gripper_width is None:
-            rospy.sleep(1.0)
-
-        was_relaxed = self.relaxed
-        if was_relaxed:
-            self.freeze()
-
-        self.program.save_arm_state(self.last_pose)
-        # TODO: assumption here, I am not grasping at the beginning...
-        self.program.save_gripper_state(pp.GripperState(self.last_gripper_width, 0.0))
-
-        if was_relaxed:
-            self.relax()
+        if not self.robotless_debug:
+            while self.last_pose is None or self.last_gripper_width is None:
+                rospy.sleep(1.0)
+
+            was_relaxed = self.relaxed
+            if was_relaxed:
+                self.freeze()
+
+            self.program.save_arm_state(self.last_pose)
+            # TODO: assumption here, I am not grasping at the beginning...
+            self.program.save_gripper_state(pp.GripperState(self.last_gripper_width, 0.0))
+
+            if was_relaxed:
+                self.relax()
+        else:
+            self.last_pose = np.random.uniform(0, 1, 3)
```

Comment:
Cool, just please document why there are sudden random samples here.

`# Here we draw random values to fake the readings from the robot during robotless debug`